tests/IdentifyServiceProviderTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Unicodeveloper\Identify\Test;
+
+use Mockery\MockInterface;
+use PHPUnit_Framework_TestCase;
+use Unicodeveloper\Identify\IdentifyServiceProvider;
+
+class IdentifyServiceProviderTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var MockInterface
+     */
+    protected $mockApp;
+
+    /**
+     * @var IdentifyServiceProvider
+     */
+    protected $provider;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->mockApp = \Mockery::mock();
+        $this->provider = new IdentifyServiceProvider($this->mockApp);
+    }
+
+    /**
+     * Test that it provides the correct service name
+     */
+    public function testRegister()
+    {
+        $this->mockApp->shouldReceive('bind')->once()->withAnyArgs()->once();
+
+        $this->provider->register();
+    }
```

Comment:
Check this out for testing the register method of the Service provider. 

Mind you our `$this->mockApp` would no longer be an anonymous mock. It will mock the ServiceProvider in the `setUp` method like so:

`$this->mockApp = m::mock('Unicodeveloper\Identify\IdentifyServiceProvider', [$this->app]);`

We are also mocking the Laravel Ioc Container itself with `$this->app = m::mock('App');`

```php
$this->app->shouldReceive('bind')
          ->once()
          ->andReturnUsing(
                function ($name, $closure) use ($test) {
                    $test->assertEquals('laravel-identify', $name);
                    $test->assertInstanceOf(
                        'Unicodeveloper\Identify\Identify',
                        $closure()
                    );
                }
            );

$this->mockApp->register();
```